scout/server/app.py

```diff
@@ -88,6 +108,22 @@ def check_user():
                 login_url = url_for("public.index", next=next_url)
                 return redirect(login_url)
 
+        if USERS_LOGGER_PATH_PARAM in app.config:
+            if current_user.is_authenticated and session.get("consent_given") is None:
```

Comment:
Since this code is found in before_request, it would log out users every time they do anything. That's why I need that consent given session variable to be in place!